src/main/java/com/google/sps/servlets/ImageHandlerServlet.java

```diff
@@ -31,108 +32,99 @@
-  private static final String PNG_PARAMETER = ".png";
   private static final String UPLOADED_REDIRECT_PARAMETER = "/uploaded.html";
   private static final String LOGIN_REDIRECT_PARAMETER = "/userAuthPage.html";
+  private static final String CHARACTER_DESIGN_REDIRECT_PARAMETER = "/characterDesign.html";
+  private static final String CONTENT_TYPE = "text/html";
+  private static final User USER = UserServiceFactory.getUserService().getCurrentUser();
   private static final String DEFAULT_IMAGE_GS_LOCATION =
       "/gs/cs-career-step-2020.appspot.com/face.jpg";
-  private static Entity currentPlayer = currentPlayer();
-  private static boolean isLoggedIn = UserServiceFactory.getUserService().isUserLoggedIn();
+  private Entity player;
+  private static final DatastoreService DATASTORE = DatastoreServiceFactory.getDatastoreService();
 
-  private static Entity currentPlayer() {
-    Entity entity = new Entity("Player");
-    try {
-      entity = PlayerDatabase.getCurrentPlayerEntity();
-    } catch (Exception e) {
-    }
-    return entity;
-  }
+  private static boolean isLoggedIn = UserServiceFactory.getUserService().isUserLoggedIn();
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String imageUrl;
-    try {
-      if (!isLoggedIn) {
-        response.sendRedirect(LOGIN_REDIRECT_PARAMETER);
-      } else {
-        // Get the displayName entered by the user.
-        String displayName = request.getParameter(DISPLAY_NAME_PARAMETER);
-        // Get the imageName to check if a file was uploaded.
-        String imageName = request.getParameter(IMAGE_PARAMETER);
-        if (imageName == null) { // Default image path
-          BlobKey defaultBlobKey = blobstoreService.createGsBlobKey(DEFAULT_IMAGE_GS_LOCATION);
-          imageUrl = getFileUrl(defaultBlobKey);
-        } else {
-          // Get the URL of the image that the user uploaded to Blobstore.
-          BlobKey uploadedBlobKey = createUploadedBlobKey(request, IMAGE_PARAMETER);
-          imageUrl = getFileUrl(uploadedBlobKey);
-        }
-        // Assign imageUrl to current player
-        PlayerDatabase.setEntityImageID(imageUrl);
+    String imageBlobKeyString;
+    String displayName = request.getParameter(MESSAGE_PARAMETER);
 
-        // Assign displayName to current player
-        PlayerDatabase.setEntityDisplayName(displayName);
-      }
-      response.sendRedirect(UPLOADED_REDIRECT_PARAMETER);
-    } catch (Exception e) {
+    UserService userService = UserServiceFactory.getUserService();
+    if (!isLoggedIn) {
+      response.sendRedirect(LOGIN_REDIRECT_PARAMETER);
+      return;
+    }
+    imageBlobKeyString = createUploadedBlobKey(request, IMAGE_PARAMETER);
+    if (imageBlobKeyString.equals("false")) {
+      response.sendRedirect(CHARACTER_DESIGN_REDIRECT_PARAMETER);
+      return;
     }
+
+    player = new Entity(PLAYER_QUERY_PARAMETER);
```

Comment:
In PlayerDatabase, functionality relies on querying players from the database by the userId. Is the userId ever set as a property for the player entity here?

Reply:
fixed!